azure/deploy-to-azure/function_template.json

```diff
@@ -153,7 +153,7 @@
   return [metadata, record];\r\n    }\r\n\r\n    getDefenderForCloudLogType(record) {\r\n        return record.type || record.Type;\r\n    }\r\n\r\n    isDefenderForCloudLog(record) {\r\n        const productName = record.ProductName;\r\n        const type = this.getDefenderForCloudLogType(record);\r\n        return DEFENDER_FOR_CLOUD_PRODUCTS.includes(productName) || DEFENDER_FOR_CLOUD_RESOURCE_TYPES.includes(type);\r\n    }\r\n\r\n    removeWhitespaceFromKeys(obj) {\r\n        \/\/ remove white space from the keys of an object\r\n        var newObj = {};\r\n        for (const [key, value] of Object.entries(obj)) {\r\n            newObj[key.replaceAll(' ', '')] = value;\r\n        }\r\n        return newObj;\r\n    }\r\n\r\n    extractMetadataFromDefenderLog(record) {\r\n        var metadata = { tags: [], source: 'defender-for-cloud', service: '' };\r\n        const productName = record.ProductName;\r\n        const type = this.getDefenderForCloudLogType(record);\r\n\r\n        if (DEFENDER_FOR_CLOUD_PRODUCTS.includes(productName)) {\r\n            metadata.service = 'SecurityAlerts';\r\n            const extendedProperties = record.ExtendedProperties || {};\r\n            record.ExtendedProperties = this.removeWhitespaceFromKeys(extendedProperties);\r\n        } else if ([SECURITY_ASSESSMENTS, SECURITY_COMPLIANCE_ASSESSMENTS].includes(type)) {\r\n            metadata.service = 'SecurityRecommendations';\r\n        } else if (type === SECURITY_SUB_ASSESSMENTS) {\r\n            metadata.service = 'SecurityFindings';\r\n        } else if ([SECURITY_SCORES, SECURITY_SCORE_CONTROLS].includes(type)) {\r\n            metadata.service = 'SecureScore';\r\n        } else {\r\n            metadata.service = 'defender-for-cloud'\r\n        }\r\n        return [metadata, record];\r\n    }\r\n}\r\n\r\nmodule.exports = async function(context, eventHubMessages) {\r\n    if (!DD_API_KEY || DD_API_KEY === '<DATADOG_API_KEY>') {\r\n        context.log.error(\r\n            'You must configure your API key before starting this function (see ## Parameters section)'\r\n        );\r\n        return;\r\n    }\r\n    try {\r\n        var handler = new EventhubLogHandler(context);\r\n        var parsedLogs = handler.handleLogs(eventHubMessages);\r\n    } catch (err) {\r\n        context.log.error('Error raised when parsing logs: ', err);\r\n        throw err;\r\n    }\r\n    var results = await new HTTPClient(context).sendAll(parsedLogs);\r\n\r\n    if (results.every(v => v === true) !== true) {\r\n        context.log.error(\r\n            'Some messages were unable to be sent. See other logs for details.'\r\n        );\r\n    }\r\n};\r\n\r\nmodule.exports.forTests = {\r\n    EventhubLogHandler,\r\n    Scrubber,\r\n    ScrubberRule,\r\n    Batcher,\r\n    constants: {\r\n        STRING,\r\n        STRING_ARRAY,\r\n        JSON_OBJECT,\r\n        JSON_ARRAY,\r\n        BUFFER_ARRAY,\r\n        JSON_STRING,\r\n        JSON_STRING_ARRAY,\r\n        INVALID\r\n    }\r\n};"
```

Comment:
This was produced by running the Javascript in index.js through a JSON escaping program. The slashes and newlines need to be escaped for the function to deploy properly.